doc/source/serve/handle-guide.md

```diff
@@ -2,74 +2,59 @@
 
 # ServeHandle: Calling Deployments from Python
 
-Ray Serve enables you to query models both from HTTP and Python. This feature
-enables seamless [model composition](serve-model-composition-guide). You can
-get a `ServeHandle` corresponding to deployment, similar how you can
-reach a deployment through HTTP via a specific route. When you issue a request
-to a deployment through `ServeHandle`, the request is load balanced across
-available replicas in the same way an HTTP request is.
-
-To call a Ray Serve deployment from python, use {mod}`Deployment.get_handle <ray.serve.api.Deployment>`
-to get a handle to the deployment, then use
-{mod}`handle.remote <ray.serve.handle.RayServeHandle.remote>` to send requests
-to that deployment. These requests can pass ordinary args and kwargs that are
-passed directly to the method. This returns a Ray `ObjectRef` whose result
-can be waited for or retrieved using `ray.wait` or `ray.get`.
-
-```python
-@serve.deployment
-class Deployment:
-    def method1(self, arg):
-        return f"Method1: {arg}"
-
-    def __call__(self, arg):
-        return f"__call__: {arg}"
-
-Deployment.deploy()
-
-handle = Deployment.get_handle()
-ray.get(handle.remote("hi")) # Defaults to calling the __call__ method.
-ray.get(handle.method1.remote("hi")) # Call a different method.
-```
+ServeHandle can be used to invoke your Serve deployment and deployment graphs using Python API.
+
+This is particularly useful for two use cases:
+Calling deployments dynamically within the deployment graph.
+Iterating and testing your application in Python.
+
+To use the ServeHandle, use {mod}`handle.remote <ray.serve.handle.RayServeHandle.remote>` to send requests to that deployment.
+These requests can pass ordinary args and kwargs that are passed directly to the method. This returns a Ray `ObjectRef` whose result can be waited for or retrieved using `await` or `ray.get`.
+
+Conceptually, ServeHandle is a client side load balancer. It has the ability to route requests to any replicas of a given deployment. It also performs buffering internally so it won’t overwhelm the replicas. The number of requests buffered is used to inform autoscaler to scale up the number of requests.
```

Comment:
```suggestion
Conceptually, ServeHandle is a client-side load balancer. It has the ability to route requests to any replicas of a given deployment. It also performs buffering internally to avoid overwhelming the replicas. The number of requests buffered is used to inform autoscaling.
```